pages/checkoutPage.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { expect, Locator, Page } from "@playwright/test";
+import { format } from "date-fns";
+
+export class CheckoutPage {
+  billingStreet: Locator;
+  billingCity: Locator;
+  billingState: Locator;
+  billingPostCode: Locator;
+  billingCountry: Locator;
+
+  paymentMethod: Locator;
+  cardNumber: Locator;
+  expirationDate: Locator;
+  cvv: Locator;
+  cardHolder: Locator;
+  confirmButton: Locator;
+  paymentSuccessMessage: Locator;
+  paymentSuccessOrder: Locator;
+  productTitle: Locator;
+  unitPrice: Locator;
+  totalPrice: Locator;
+  proceedToCheckout1: Locator;
+  proceedToCheckout2: Locator;
+  proceedToCheckout3: Locator;
+  userLoggedInMessage: Locator;
+
+  constructor(private page: Page) {
+    this.billingStreet = page.getByTestId("street");
+    this.billingCity = page.getByTestId("city");
+    this.billingState = page.getByTestId("state");
+    this.billingCountry = page.getByTestId("country");
+    this.billingPostCode = page.getByTestId("postal_code");
+    this.paymentMethod = page.getByTestId("payment-method");
+    this.cardNumber = page.getByTestId("credit_card_number");
+    this.expirationDate = page.getByTestId("expiration_date");
+    this.cvv = page.getByTestId("cvv");
+    this.cardHolder = page.getByTestId("card_holder_name");
+    this.confirmButton = page.getByTestId("finish");
+    this.paymentSuccessMessage = page.getByText("Payment was successful");
+    this.paymentSuccessOrder = page.getByText(
+      "Thanks for your order! Your invoice number is"
+    );
+    this.productTitle = page.getByTestId("product-title");
+    this.unitPrice = page.getByTestId("product-price");
+    this.totalPrice = page.getByTestId("cart-total");
+    this.proceedToCheckout1 = page.getByTestId("proceed-1");
+    this.proceedToCheckout2 = page.getByTestId("proceed-2");
+    this.proceedToCheckout3 = page.getByTestId("proceed-3");
+    this.userLoggedInMessage = page.getByText("you are already logged in", {
+      exact: false,
+    });
+  }
+
+  async fillBillingAddress() {
+    await this.billingStreet.fill("Main St, 123");
```

Comment:
Let's move the address details to the test data. And pass the arguments into the function. The same below